code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -128,12 +133,15 @@
 		return
 	return ..()
 
-/obj/item/baton/apply_hit_effect(mob/living/target, mob/living/user, var/hit_zone)
+/obj/item/baton/apply_hit_effect(mob/living/target, mob/living/user, var/hit_zone, var/thrown = FALSE)
 	if(isrobot(target))
 		return ..()
 
 	var/agony = agonyforce
 	var/stun = stunforce
+	var/throwfail = 0
+	if(thrown)
+		throwfail = user.skill_fail_prob(SKILL_COMBAT, 15, no_more_fail = SKILL_EXPERT)
```

Comment:
Idle thought but it might be fun to make thrown knives/sharp objects stab people if they pass this check.